src/models/rule_store.rs

```diff
@@ -143,19 +143,58 @@ impl RuleStore {
   /// Note that `WalkDir` traverses the directory with parallelism.
   /// If all the global rules have no holes (i.e. we will have no grep patterns), we will try to find a match for each global rule in every file in the target.
   pub(crate) fn get_relevant_files(
-    &self, path_to_codebase: &str, include: &Vec<Pattern>, exclude: &Vec<Pattern>,
+    &self, paths_to_codebase: &Vec<String>, include: &Vec<Pattern>, exclude: &Vec<Pattern>,
   ) -> HashMap<PathBuf, String> {
-    let _path_to_codebase = Path::new(path_to_codebase).to_path_buf();
-
-    //If the path_to_codebase is a file, then execute piranha on it
-    if _path_to_codebase.is_file() {
-      return HashMap::from_iter([(
-        _path_to_codebase.clone(),
-        read_file(&_path_to_codebase).unwrap(),
-      )]);
+    let mut candidate_files: HashMap<PathBuf, String> = HashMap::new();
+
+    for p2codebase in paths_to_codebase {
+      if !Path::new(p2codebase).exists() {
+        panic!("Path to codebase does not exist: {}", p2codebase);
```

Comment:
Are we testing for this error condition anywhere in our test suite? Seems new and not in the one newly added test case.

Reply:
added a test